common/djangoapps/dark_lang/tests.py

```diff
@@ -229,78 +228,120 @@ def test_partial_match_es_la(self, latin_america_code):
-            self.process_request(preview_lang='rel', language_session_key='notrel')
+            response.client.session
         )
 
     def test_preview_lang_with_dark_language(self):
-        self.assertSessionLangEquals(
+        self.post_set_preview_lang('unrel')
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             'unrel',
-            self.process_request(preview_lang='unrel')
+            response.client.session
         )
 
-        self.assertSessionLangEquals(
+        # Test a clear and then a set of the preview language
+        self.post_clear_preview_lang()
+        self.post_set_preview_lang('unrel')
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             'unrel',
-            self.process_request(preview_lang='unrel', language_session_key='notrel')
+            response.client.session
         )
 
-    def test_clear_lang(self):
-        self.assertSessionLangEquals(
+        # When posting an empty preview_language the currently set language should not change
+        self.post_clear_preview_lang()
+        self.post_set_preview_lang(' ')
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             UNSET,
-            self.process_request(clear_lang=True)
+            response.client.session
         )
 
-        self.assertSessionLangEquals(
+    def test_clear_lang(self):
+        # Clear a language when no language was set
+        self.post_clear_preview_lang()
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             UNSET,
-            self.process_request(clear_lang=True, language_session_key='rel')
+            response.client.session
         )
 
-        self.assertSessionLangEquals(
+        # Set a language and clear it to ensure the clear is working as expected
+        self.post_set_preview_lang('notclear')
+        self.post_clear_preview_lang()
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             UNSET,
-            self.process_request(clear_lang=True, language_session_key='unrel')
+            response.client.session
         )
 
     def test_disabled(self):
         DarkLangConfig(enabled=False, changed_by=self.user).save()
 
         self.assertAcceptEquals(
             'notrel;q=0.3, rel;q=1.0, unrel;q=0.5',
-            self.process_request(accept='notrel;q=0.3, rel;q=1.0, unrel;q=0.5')
+            self.process_middleware_request(accept='notrel;q=0.3, rel;q=1.0, unrel;q=0.5')
         )
 
-        self.assertSessionLangEquals(
+        self.post_clear_preview_lang()
+        self.assert_request_session_lang_equals(
             'rel',
-            self.process_request(clear_lang=True, language_session_key='rel')
+            self.process_middleware_request(language_session_key='rel')
         )
 
-        self.assertSessionLangEquals(
+        self.post_clear_preview_lang()
```

Comment:
I'm having trouble understanding the general workflow of these tests. For example, when do you call assert_session_lang_equals vs. assert_request_session_lang_equals? And in this test, why clear, then set to "rel", then ultimately expect a language of "unrel" when calling process_middleware_request directly with "unrel"?

More comments would be helpful, and perhaps also split up the test points so each one is only focused on a single test case.